src/it/java/teammates/it/storage/sqlapi/NotificationDbIT.java

```diff
@@ -0,0 +1,88 @@
+                .withStartTime(Instant.parse("2011-01-01T00:00:00Z"))
+                .withEndTime(Instant.parse("2099-01-01T00:00:00Z"))
+                .withStyle(NotificationStyle.DANGER)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withMessage("<p>Deprecation happens in three minutes</p>")
+                .build();
+
+        notificationsDb.createNotification(newNotification);
+
+        UUID notificationId = newNotification.getNotificationId();
+        Notification actualNotification = notificationsDb.getNotification(notificationId);
+        verifyEquals(newNotification, actualNotification);
+
+        ______TS("failure: create notification that already exists, exception thrown");
+        Notification identicalNotification = new Notification.NotificationBuilder("A deprecation note")
+                .withStartTime(Instant.parse("2011-01-01T00:00:00Z"))
+                .withEndTime(Instant.parse("2099-01-01T00:00:00Z"))
+                .withStyle(NotificationStyle.DANGER)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withMessage("<p>Deprecation happens in three minutes</p>")
+                .build();
+        identicalNotification.setNotificationId(notificationId);
+
+        assertNotSame(newNotification, identicalNotification);
+        EntityAlreadyExistsException ex = assertThrows(EntityAlreadyExistsException.class,
+                () -> notificationsDb.createNotification(identicalNotification));
+        assertEquals(ex.getMessage(), "Trying to create an entity that exists: " + identicalNotification.toString());
+
+        ______TS("failure: invalid parameter, start time is before end time");
+        Notification invalidNotification1 = new Notification.NotificationBuilder("A deprecation note")
+                .withStartTime(Instant.parse("2011-02-01T00:00:00Z"))
+                .withEndTime(Instant.parse("2011-01-01T00:00:00Z"))
+                .withStyle(NotificationStyle.DANGER)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withMessage("<p>Deprecation happens in three minutes</p>")
+                .build();
+
+        assertThrows(InvalidParametersException.class, () -> notificationsDb.createNotification(invalidNotification1));
+
+        ______TS("failure: empty title");
+        Notification invalidNotification2 = new Notification.NotificationBuilder("")
+                .withStartTime(Instant.parse("2011-01-01T00:00:00Z"))
+                .withEndTime(Instant.parse("2099-01-01T00:00:00Z"))
+                .withStyle(NotificationStyle.DANGER)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withMessage("<p>Deprecation happens in three minutes</p>")
+                .build();
+
+        assertThrows(InvalidParametersException.class, () -> notificationsDb.createNotification(invalidNotification2));
+
+        ______TS("failure: empty message");
```

Comment:
For integration tests, try not to test too much. There's quite significant costs to each additional test (running time, dev time, maintainability, etc.). Especially since in this context the devs are the ones that maintain both integration & unit tests, we can skip some redundant tests.

Not a "law", but in general, we want to avoid testing things that do not test the integration between modules/components and are already covered by our unit test. Let the unit tests, which have much less associated costs, check for these (usually edge) cases. In this case, we only want to test the integration between db layer and db/hibernate itself, so we can consider not testing for edge cases that do not interact with the db.